### PR TITLE
YD-165 implement goals controller, YD-46 embed goals in user

### PR DESCRIPTION
**Pull request by aristotelos**
Updating of goals not supported yet

### CODE REVIEW

File: appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy [hunk 1]
```diff
@@ -33,22 +33,20 @@ abstract class AbstractAppServiceIntegrationTest extends Specification
...
 	{
 		def richard = appService.addUser(appService.&assertUserCreationResponseDetails, "R i c h a r d", "Richard", "Quinn", "RQ",
-				"+$timestamp", ["Nexus 6"], [
-					BudgetGoal.createNoGoInstance("news"),
-					BudgetGoal.createNoGoInstance("gambling")
-				])
+				"+$timestamp", ["Nexus 6"])
 		appService.confirmMobileNumber(appService.&assertResponseStatusSuccess, richard)
+		appService.addBudgetGoal(richard, BudgetGoal.createNoGoInstance("news"))
+		appService.addBudgetGoal(richard, BudgetGoal.createNoGoInstance("gambling"))
```
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: Thanks for choosing this approach. The earlier "inline" approach where the goals were given during user creation was just a convenience. This is indeed the way to go.

File: appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy [hunk 2]
```diff
@@ -94,7 +92,7 @@ class CreateUserOnBuddyRequestTest extends AbstractAppServiceIntegrationTest
 		response.responseData.nickname == newNickname
 		response.responseData.devices.size() == 1
 		response.responseData.devices[0] == "iPhone 6"
-		response.responseData.goals.size() == 0 //TODO: updating of goals is not yet supported
+		//response.responseData._embedded._budgetGoals.size() == 0 //TODO: updating of goals is not yet supported
```
Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy]: Why is this commented? I would expect Bob to still have zero goals, so this line can be written as 
```
!response.responseData._embedded?._budgetGoals?.size
```

File: appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy (hunk 1 above)
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: There was no other way actually to do this nicely with HAL. At least Spring cannot deserialize a JSON object containing _embedded items I guess :)

Comment (Bert-R) [inline appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy]: I want to give that a try later. I'm still hopeful :)

**Comment by Bert-R** [comment]
Besides one comment on ``CreateUserOnBuddyRequestTest``, it's fine with me.

File: appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy (hunk 2 above)
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy]: Agreed, thanks for the suggestion

**Comment by aristotelos** [comment]
Thanks for the fast review! Have fixed the TODO.

**Pull request merged**